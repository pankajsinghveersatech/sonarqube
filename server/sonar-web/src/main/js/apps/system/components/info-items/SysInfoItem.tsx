/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { map } from 'lodash';
import * as React from 'react';
import { translate } from '../../../../helpers/l10n';
import { HEALTH_FIELD, STATE_FIELD } from '../../utils';
import HealthItem from './HealthItem';

export interface Props {
  name: string;
  value: T.SysInfoValue;
}

export default function SysInfoItem({ name, value }: Props) {
  if (name === HEALTH_FIELD || name === STATE_FIELD) {
    return <HealthItem className="no-margin" health={value as T.HealthType} />;
  }
  if (value instanceof Array) {
    return <code>{JSON.stringify(value)}</code>;
  }
  switch (typeof value) {
    case 'boolean':
      return <>{translate(value ? 'yes' : 'no')}</>;
    case 'object':
      return (
        <table className="data">
          <tbody>
            {map(value, (v, n) => (
              <tr key={n}>
                <td className="thin nowrap">{n}</td>
                <td>
                  <SysInfoItem name={n} value={v} />
                </td>
              </tr>
            ))}
          </tbody>
        </table>
      );
    default:
      return <code>{value}</code>;
  }
}
