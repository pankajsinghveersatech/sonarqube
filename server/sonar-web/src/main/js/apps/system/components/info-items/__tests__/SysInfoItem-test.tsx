/*
 * SonarQube
 * Copyright (C) 2009-2021 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
import { shallow } from 'enzyme';
import * as React from 'react';
import HealthItem from '../HealthItem';
import SysInfoItem, { Props } from '../SysInfoItem';

it('should render correctly', () => {
  expect(shallowRender('/some/path/as/an/example')).toMatchSnapshot('string');
  expect(shallowRender({ foo: 'Far', bar: { a: 1, b: 'b' }, baz: true })).toMatchSnapshot('object');
  expect(shallowRender(true)).toMatchSnapshot('true');
  expect(shallowRender(false)).toMatchSnapshot('false');
});

it('should render health item', () => {
  expect(
    shallowRender('GREEN', 'Health')
      .find(HealthItem)
      .prop('health')
  ).toBe('GREEN');
});

function shallowRender(value: Props['value'], name: Props['name'] = 'foo') {
  return shallow(<SysInfoItem name={name} value={value} />);
}
