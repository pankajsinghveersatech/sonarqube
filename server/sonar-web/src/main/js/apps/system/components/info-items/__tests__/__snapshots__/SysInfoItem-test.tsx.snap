// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly: false 1`] = `
<Fragment>
  no
</Fragment>
`;

exports[`should render correctly: object 1`] = `
<table
  className="data"
>
  <tbody>
    <tr
      key="foo"
    >
      <td
        className="thin nowrap"
      >
        foo
      </td>
      <td>
        <SysInfoItem
          name="foo"
          value="Far"
        />
      </td>
    </tr>
    <tr
      key="bar"
    >
      <td
        className="thin nowrap"
      >
        bar
      </td>
      <td>
        <SysInfoItem
          name="bar"
          value={
            Object {
              "a": 1,
              "b": "b",
            }
          }
        />
      </td>
    </tr>
    <tr
      key="baz"
    >
      <td
        className="thin nowrap"
      >
        baz
      </td>
      <td>
        <SysInfoItem
          name="baz"
          value={true}
        />
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`should render correctly: string 1`] = `
<code>
  /some/path/as/an/example
</code>
`;

exports[`should render correctly: true 1`] = `
<Fragment>
  yes
</Fragment>
`;
